src/components/FileUpload.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import '@/css/FileUpload.css';
+
+interface FileUploadProps {
+  uploadUrl: string;
+}
+
+const FileUpload: React.FC<FileUploadProps> = ({ uploadUrl }) => {
+  const [showModal, setShowModal] = useState(false);
+  const [selectedFiles, setSelectedFiles] = useState<File[]>([]);
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files) {
+      const newFiles = Array.from(event.target.files);
+      setSelectedFiles((prevFiles) => [...prevFiles, ...newFiles]);
+    }
+  };
+
+  const handleUpload = async () => {
+    // Once POST API req is out -- Currently WIP so cannot make any progress here
+    if (selectedFiles.length > 0) {
+      const formData = new FormData();
+      selectedFiles.forEach((file) => {
+        formData.append('files', file);
+      });
+
+      try {
+        const response = await fetch(uploadUrl, {
+          method: 'POST',
+          body: formData,
+        });
+
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
```

Comment:
Are we catching this anywhere? From a glance it looks like we are not. If we don't catch it'll just crash. For now while it is in WIP, it's fine, but we should definitely work on another component which displays a popup for errors/info which we can configure with passed in props. We can then use that for most errors.